### ISSUE TITLE
Uncaught SyntaxError: Failed to execute 'addColorStop' on 'CanvasGradient'

### ISSUE THREAD

**Issue by eikooc** [thomaslindstrom/color-picker#120]
Color picker doesn't work as of latest version.

This syntax error comes no matter what.

These two are slightly different but it seems like the same issue.

```
Uncaught SyntaxError: Failed to execute 'addColorStop' on 'CanvasGradient': The value provided ('rgba(-Infinity, NaN, NaN, 1)') could not be parsed as a color.
```

```
Uncaught SyntaxError: Failed to execute 'addColorStop' on 'CanvasGradient': The value provided ('rgba(NaN, NaN, NaN, 1)') could not be parsed as a color.
```

This has been tried on multiple computers

**Comment by thomaslindstrom** [comment]
Hi, @eikooc. What version of Atom and what version of `color-picker` are you on?

**Comment by protetore** [comment]
Same error here, my Atom version is 0.204.0 and my color-picker version is 2.0.7.

**Comment by MarcFowler** [comment]
I'm on Atom 0.210.0 and color-picker version 2.0.7 and I get the same thing no matter what I try:

```
Uncaught SyntaxError: Failed to execute 'addColorStop' on 'CanvasGradient': The value provided ('rgba(-Infinity, NaN, Infinity, 1)') could not be parsed as a color.
```

**Stack trace**
```
Error: Failed to execute 'addColorStop' on 'CanvasGradient': The value provided ('rgba(-Infinity, NaN, Infinity, 1)') could not be parsed as a color.
  at Error (native)
  at Object._this.canvas.render (/Users/marc/.atom/packages/color-picker/lib/extensions/Alpha.coffee:100:35)
  at /Users/marc/.atom/packages/color-picker/lib/extensions/Alpha.coffee:109:29
  at Object.emit (/Users/marc/.atom/packages/color-picker/lib/modules/Emitter.coffee:11:23)
  at Object.emitColorChanged (/Users/marc/.atom/packages/color-picker/lib/extensions/Saturation.coffee:24:22)
  at /Users/marc/.atom/packages/color-picker/lib/extensions/Saturation.coffee:197:61
  at Object.emit (/Users/marc/.atom/packages/color-picker/lib/modules/Emitter.coffee:11:23)
  at Object.emitSelectionChanged (/Users/marc/.atom/packages/color-picker/lib/extensions/Saturation.coffee:18:22)
  at Object._this.control.setSelection (/Users/marc/.atom/packages/color-picker/lib/extensions/Saturation.coffee:186:43)
  at /Users/marc/.atom/packages/color-picker/lib/extensions/Saturation.coffee:194:30
  at Object.emit (/Users/marc/.atom/packages/color-picker/lib/modules/Emitter.coffee:11:23)
  at Object.emitInputColor (/Users/marc/.atom/packages/color-picker/lib/ColorPicker-view.coffee:232:22)
  at Object.open (/Users/marc/.atom/packages/color-picker/lib/ColorPicker-view.coffee:309:23)
  at atom-text-editor.<anonymous> (/Users/marc/.atom/packages/color-picker/lib/ColorPicker.coffee:42:68)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:238:29)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:180:20)
```

**Comment by MarcFowler** [comment]
fwiw I'm selecting hex colours (`#933432`) and then `cmd+shift+c`ing to bring up the colour selector, but it just throws that error.

**Comment by thomaslindstrom** [comment]
Hi, @MarcFowler. What happens when you go into `color-picker` settings and disable `Serve a random color on open`? Is it still there?

**Comment by MarcFowler** [comment]
Actually it seems to be working perfectly in 2.0.7! :+1: I'd close this as I've not been able to repeat this (I'm a pretty heavy user of it!).

**Comment by heidar-ice** [comment]
Well, not working for me. And disabling random color doesn't help.

**Comment by thomaslindstrom** [comment]
@heidar-ice With the `color-picker` updated to the newest version (`v2.0.8`, published now), Atom reloaded, initialised and package activated, could you try opening the inspector (`CMD+ALT+I`) and then in `Console` tell me what you get when you enter this?

```
document.querySelector('.ColorPicker-body-saturation').getClientRects()[0]
```

**Comment by heidar-ice** [comment]
@thomaslindstrom Went to check that for you and suddenly it's working again even on 2.0.7. Did nothing special except rebooting the VM and reopening folder at new location. You still need that output?

**Comment by thomaslindstrom** [comment]
@heidar-ice Not really, no. Just knowing this bug is sporadic and not constant is a relief. I might have fixed it in `v2.0.8`, too. Thanks!
